addons/sale_expense/models/account_move.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _reverse_moves(self, default_values_list=None, cancel=False):
+        # EXTENDS sale
+        self.expense_sheet_id._sale_expense_reset_sol_quantities()
+        return super()._reverse_moves(default_values_list, cancel)
```

Comment:
What happens in case of a partial reverse?

Reply:
Because we have to work with the quantities it will work as if it's a full one. Should be improved in the future but this is just the stepping stone PR